testing/trino-tests/src/test/java/io/trino/tests/AbstractTestEngineOnlyQueries.java

```diff
@@ -5810,10 +5810,10 @@ public void testLargePivot()
     @Test
     public void testPivotExceedingMaximumArraySize()
     {
-        assertQueryFails(pivotQuery(255), "Too many arguments for array constructor");
+        assertQueryFails(pivotQuery(255, false), "Too many arguments for array constructor");
     }
 
-    private static String pivotQuery(int columnsCount)
+    private static String pivotQuery(int columnsCount, boolean interpretable)
```

Comment:
Not sure I understand why this is needed

Reply:
`interpretable` is a way to decide whether the array will be created in `ExpressionInterpreter`.
If all of the array elements can be evaluated, then the Interpreter creates the array and it doesn't fail even if the array is very big. 
I didn't find out though why this test passed before my change.

Reply:
I investigated a bit and here is what I found:
- before, the array constructor failed in `ExpressionInterpreter`
- after I added error suppression, it stopped to fail in `ExpressionInterpreter`, and did not fail at all, because it missed the size check in `ArrayConstructor`. **Probably it is a bug.**
- then I added `0/0` to the arrays and it started failing again with "Too many arguments for array constructor". But it was neither in `ExpressionInterpreter` nor in `ArrayConstructor`. It was **in the `ExpressionAnalyzer`** invoked by `DesugarLikeRewrite`.

I think we need an unconditional size check in the `ArrayConstructor`. We cannot rely on a chance to hit one of the checks we've got here and there.